src/components/modals/login-modal/LoginModal.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useState } from 'react';
+import { Input, Modal, Typography } from 'antd';
+import styled from 'styled-components';
+import './login-modal.less';
+import { login } from '../../../auth/ducks/thunks';
+import { connect, useDispatch } from 'react-redux';
+import Text from 'antd/es/typography/Text';
+import { Link } from 'react-router-dom';
+import { C4CState } from '../../../store';
+import { AsyncRequest, AsyncRequestKinds } from '../../../utils/asyncRequest';
+import { TokenPayload } from '../../../auth/ducks/types';
+
+interface LoginModalProps {
+  showLoginModal: boolean;
+  onCloseLoginModal: () => void;
+}
+
+interface StateProps {
+  tokens: AsyncRequest<TokenPayload, any>;
+}
+
+enum ModalContent {
+  LoginContent,
+  ForgotPassword,
+  ResetPassword,
+}
+
+const StyledModal = styled(Modal)`
+  horiz-align: center;
+`;
+
+const ContentDiv = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+`;
+
+const ForgotPasswordLinkButton = styled(Typography.Link)`
+  color: black;
+  white-space: nowrap;
+  background-color: transparent;
+  border: transparent;
+  padding: 0px 0px 0px 0px;
+`;
+
+const ForgotPasswordText = styled(Text)`
+  color: #000000;
+`;
+
+const EmailInput = styled(Input)`
+  margin-top: 10px;
+  margin-bottom: 10px;
+  width: 272px;
+`;
+
+const PasswordInput = styled(Input.Password)`
+  margin-bottom: 10px;
+  width: 272px;
+`;
+
+const LoginModal: React.FC<LoginModalProps & StateProps> = ({
```

Comment:
Great interface for the functionality we want. At some point, we may consider passing tokens in as a "true" prop, ie passed from a parent container rather than connecting it to the Redux store itself.